plugins/blockvault_client_plugin/postgres_backend.cpp

```diff
@@ -0,0 +1,196 @@
+#include "postgres_backend.hpp"
+#include <boost/filesystem.hpp>
+#include <fstream>
+#include <iostream>
+#include <limits.h>
+#include <unistd.h>
+
+namespace eosio {
+namespace blockvault {
+
+postgres_backend::postgres_backend(const std::string& options)
+    : conn(options) {
+   pqxx::work w(conn);
+   w.exec("CREATE TABLE IF NOT EXISTS BlockData (watermark_bn bigint, watermark_ts bigint, lib bigint, "
```

Comment:
I'm surprised not to see indices over some of the selected columns like `watermark_(bn|ts)` `lib` and `block_id`, was that unnecessary or reserved for future perf analysis?

Reply:
This is what I got from https://www.w3schools.com/sql/sql_create_index.asp
```
Note: Updating a table with indexes takes more time than updating a table without (because the indexes also need an update). So, only create indexes on columns that will be frequently searched against.
```

Base on the above statement, I have double if it's worthwhile to create indexes.

Reply:
If we are not seeing performance issues at this time, then we can leave it as-is.  It is true that updates/inserts are more costly with indices *however* our updates have dependent queries associated with them to enforce our guarantees.

We should profile this with a substantially sized database and see if the trade of certain indices improves update time by reducing those dependent read-times more than it increases the time to write to indices.  

There is also another option where we do some fancy materialized view of the current highest watermarks BUT we should not engage in these optimizations until we have baseline performance numbers and a plan to iterate.

Reply:
resolving this for now as it was mostly a thing to pay attention to in our next iteration.